src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx

```diff
@@ -38,51 +40,94 @@ const FilterGroup: React.FC<FilterGroupProps> = ({
 
   const [newFilterName, setNewFilterName] = useState(name);
   const [isEditing, setIsEditing] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+  const editFieldRef = useRef<HTMLInputElement>(null);
 
   const resetEditState = () => {
     setIsEditing(false);
     setNewFilterName(name);
   };
 
+  const handleSubmit = () => {
+    if (isValid) {
+      editFilter("name", newFilterName, filter);
+      resetEditState();
+    }
+  };
+
+  useEffect(() => {
+    if (isEditing) {
+      editFieldRef.current?.focus();
+    }
+  }, [isEditing, editFieldRef]);
+
   return (
     <FilterContainer data-cy={dataCy}>
       <FilterHeader>
         <Badge>FILTER</Badge>
         <IconButtonContainer>
-          <IconButton
-            aria-label="Edit filter"
-            onClick={() => setIsEditing(true)}
+          <Tooltip
+            popoverZIndex={zIndex.tooltip}
+            trigger={
+              <IconButton
+                aria-label="Edit filter"
+                onClick={() => setIsEditing(true)}
+              >
+                <Icon fill={gray.base} glyph="Edit" />
+              </IconButton>
+            }
           >
-            <Icon fill={gray.base} glyph="Edit" />
-          </IconButton>
-          <IconButton
-            aria-label={visible ? "Hide filter" : "Show filter"}
-            onClick={() => editFilter("visible", !visible, filter)}
+            Edit filter
+          </Tooltip>
+          <Tooltip
+            popoverZIndex={zIndex.tooltip}
+            trigger={
+              <IconButton
+                aria-label={visible ? "Hide filter" : "Show filter"}
+                onClick={() => editFilter("visible", !visible, filter)}
+              >
+                <Icon
+                  fill={gray.base}
+                  glyph={visible ? "Visibility" : "ClosedEye"}
+                />
+              </IconButton>
+            }
           >
-            <Icon
-              fill={gray.base}
-              glyph={visible ? "Visibility" : "ClosedEye"}
-            />
-          </IconButton>
-          <IconButton
-            aria-label="Delete filter"
-            onClick={() => deleteFilter(name)}
+            {visible ? "Hide filter" : "Show filter"}
+          </Tooltip>
+          <Tooltip
+            popoverZIndex={zIndex.tooltip}
+            trigger={
+              <IconButton
+                aria-label="Delete filter"
+                onClick={() => deleteFilter(name)}
+              >
+                <Icon fill={gray.base} glyph="X" />
+              </IconButton>
+            }
           >
-            <Icon fill={gray.base} glyph="X" />
-          </IconButton>
+            Delete filter
+          </Tooltip>
         </IconButtonContainer>
       </FilterHeader>
 
       {isEditing ? (
         <>
           <StyledTextInput
+            ref={editFieldRef}
```

Comment:
As an alternative, I think you could remove the `ref` logic and use the native property `autoFocus`

Reply:
Wow great one!